resources/qml/TableView.qml

```diff
@@ -1,67 +1,217 @@
-            implicitWidth: UM.Theme.getSize("scrollbar").width
-            implicitHeight: UM.Theme.getSize("scrollbar").width
-            radius: width / 2
-            color: UM.Theme.getColor(styleData.pressed ? "scrollbar_handle_down" : (styleData.hovered ? "scrollbar_handle_hover" : "scrollbar_handle"))
+            return headerBar.children[column].width; //Cells get the same width as their column header.
         }
 
-        scrollBarBackground: Rectangle
+        delegate: Rectangle
         {
-            // Both implicit width and height have to be set, since the handle is used by both the horizontal and the vertical scrollbars
-            implicitWidth: UM.Theme.getSize("scrollbar").width
-            implicitHeight: UM.Theme.getSize("scrollbar").width
-            color: UM.Theme.getColor("main_background")
+            implicitHeight: Math.max(1, cellContent.height)
+
+            color: UM.Theme.getColor((tableBase.currentRow == row) ? "primary" : ((row % 2 == 0) ? "main_background" : "viewport_background"))
+
+            Label
+            {
+                id: cellContent
+                width: parent.width
+
+                text: display
+                verticalAlignment: Text.AlignVCenter
+                elide: Text.ElideRight
+                font: UM.Theme.getFont("default")
+                color: UM.Theme.getColor("text")
+            }
+            TextMetrics
+            {
+                id: cellTextMetrics
+                text: cellContent.text
+                font: cellContent.font
+                elide: cellContent.elide
+                elideWidth: cellContent.width
+            }
+            UM.TooltipArea
+            {
+                anchors.fill: parent
+
+                text: (cellTextMetrics.elidedText == cellContent.text) ? "" : cellContent.text //Show full text in tooltip if it was elided.
+                onClicked:
+                {
+                    if(tableBase.allowSelection)
+                    {
+                        tableBase.currentRow = row; //Select this row.
+                    }
+                }
+                onDoubleClicked:
+                {
+                    tableBase.onDoubleClicked(row);
+                }
+            }
         }
 
-        // The little rectangle between the vertical and horizontal scrollbars
-        corner: Rectangle
+        Connections
         {
-            color: UM.Theme.getColor("main_background")
+            target: model
+            function onRowCountChanged()
+            {
+                tableView.contentY = 0; //When the number of rows is reduced, make sure to scroll back to the start.
+            }
         }
+    }
+
+    Connections
+    {
+        target: model
+        function onRowsChanged()
+        {
+            let first_column = model.columns[0].display;
+            if(model.rows.length > 0 && model.rows[0][first_column].startsWith("<b>")) //First item is already a section header.
```

Comment:
It would be more reliable if we don't strip spaces or check for caps, lest the user of the table puts in `<b>` themselves. This check is unreliable in theory if the user enters actual data that starts with `<b>` in the first cell, and wants to use section headers, because this check then thinks there are already section headers in the table and doesn't add them any more.

The `<b>` is added on line 207. It is always `<b>`, no spaces in front and always lowercase. We could consider using something more unique, like a marker at the start of the first cell to indicate that the cell contents were already modified (maybe `<B>` would be less common?). But if it's a unique signalling character, we have to make sure it's not a character that is visible to the user. This if-statements essentially prevents going into an infinite loop because at the end of this function it modifies the `rows` field which triggers another `onRowsChanged()` signal.

My initial attempt to fix this issue was to have a separate model that the user provides, and then an internal model - with modified cells - for display. This attempt failed however because the `TableViewColumn`s can't be copied in Qt and also can't be forwarded because the children of `TableViewModel` are restructured into an internal `columns` field which you can't assign to. So the only way I could get it to work was to have only the one model that the user provides, and modify it in-place like this. It should only be done once in theory, except that we re-use the model ourselves in places just changing the model contents when switching tabs for the profile manager.